### PR TITLE
Simplify `WorkflowRuntimeServer` stop logic

### PR DESCRIPTION
**Pull request by MetRonnie**
Plus some tidying & adding type annotations.

This was split off from #5098

**Check List**

- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] No dependency changes to both `setup.cfg` and `conda-environment.yml`.
- [x] Tests are updated
- [x] No `CHANGES.md` entry included (invisible to users)
- [x] No docs needed
- [x] PR for master only

### CODE REVIEW

File: cylc/flow/network/__init__.py [hunk 1]
```diff
@@ -112,31 +112,24 @@ class ZMQSocketBase:
...
         self.stopping = False
 
     def start(self, *args, **kwargs):
-        """Start the server/network-component.
-
-        Pass arguments to _start_
-        """
-        self._start_sequence(*args, **kwargs)
-
-    def _start_sequence(self, *args, **kwargs):
```
Comment (dwsutherland) [inline cylc/flow/network/__init__.py:139]: The legacy reason for this was the thread use to be started from here.

File: cylc/flow/network/subscriber.py [hunk 2]
```diff
@@ -49,45 +52,45 @@ class WorkflowSubscriber(ZMQSocketBase):
...
+    loop: asyncio.AbstractEventLoop
 
     def __init__(
-            self,
-            workflow: str,
-            host: str = None,
-            port: Union[int, str] = None,
-            context: object = None,
-            srv_public_key_loc: str = None,
-            topics: Iterable[bytes] = None
```
Comment (dwsutherland) [inline cylc/flow/network/subscriber.py:69]: I never really liked the double indentation .. But the linter use to pull us up otherwise.

**Review by dwsutherland — APPROVED** [review]
LGTM.. tests passing.

**Review by hjoliver — APPROVED** [review]
:+1:

**Pull request merged**